src/lib.rs

```diff
@@ -855,6 +867,59 @@ impl<I, F> StreamingIterator for TakeWhile<I, F>
     }
 }
 
+/// A streaming iterator which will continue with another operator when the first one finishes
+pub struct AndThen<I, J> {
+    it: I,
+    then: J,
+    done: bool,
+}
+
+impl<I, J> StreamingIterator for AndThen<I, J>
+    where I: StreamingIterator,
+          J: StreamingIterator<Item=I::Item>
+{
+    type Item = I::Item;
+
+    #[inline]
+    fn advance(&mut self) {
+        if !self.done {
+            self.it.advance();
+            if self.it.get().is_none() {
+                self.done = true;
+                self.then.advance();
```

Comment:
I'm not quite sure this is safe. What if `advance` has already been called. We will miss one element.

I'm not sure what I'm supposed to do though.